pkg/platform/secret.go

```diff
@@ -114,7 +117,23 @@ func ApplyPlatformSecret(
 				},
 			},
 			Data: secretPayload,
-		}, metav1.CreateOptions{})
+		}
+		if svcErr == nil {
```

Comment:
Would we not want to return an error if the service can't be found?  Also there's a [SetOwnerReference](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetOwnerReference) method which would make this look more like 

```go
if err := controllerutil.SetOwnerReference(vClusterService, secret, scheme.Scheme); err != nil {
  return err
}
```

Reply:
I initially used it, but then decided I want to set `BlockOwnerDeletion` to `false`, and in `SetOwnerReference` it's always set to true

Reply:
Not worth changing at this point, but you can also pass an option for that as `controllerutil.WithBlockOwnerDeletion(false)` after the scheme